decidim-core/app/helpers/decidim/tos_page_helper.rb

```diff
@@ -0,0 +1,95 @@
+# frozen_string_literal: true
+
+module Decidim
+  # A Helper to render:
+  # - the TOS updated announcement when redirected to the TOS page.
+  # - the Accept updated TOS form in the TOS page.
+  # - the Modal with info when refusing the updated TOS.
+  module TosPageHelper
+    # Renders the Call To Action button. Link and text can be configured
+    # per organization.
+
+    def tos_page_announcement
+      return unless current_user
+      return if current_user.tos_accepted?
+      locals = {
+        callout_class: "warning",
+        announcement: {
+          title: t("required_review.title", scope: "decidim.pages.terms_and_conditions"),
+          body: t("required_review.body", scope: "decidim.pages.terms_and_conditions")
+        }
+      }
+      html = render(partial: "decidim/shared/announcement", locals: locals).to_s
+      html += %(<div id="sticky-top-stop"></div>).html_safe
+      html.html_safe
+    end
+
+    def tos_page_sticky_form
```

Comment:
Move this method and following ones to a cell

Reply:
Sure!